src/narmvm.rs

```diff
@@ -877,6 +899,17 @@ impl NarmVM{
     pub fn get_pc_address(&self) -> u32{
         self.pc & (!1)
     }
+    pub fn external_set_reg(&mut self, reg: usize, value: u32){
+        if reg < 8 {
+            self.sreg[reg] = value;
+        }
+        else if reg < 15 {
+            self.long_registers[reg - 8] = value;
+        }
+        else if reg == 15 {
+            self.pc = value;
```

Comment:
This should not be directly settable externally, specifically should use the recent system for set_pc_thumb_address etc

Reply:
Actually maybe this is ok in an external context, though the other methods should be preferred

Reply:
I figure that it can be expected of a test writer to put the correct value in there.